### ISSUE TITLE
Version v0.11.0 of `netclient join` fails on the processor on which the Netmaker server is installed.

### ISSUE THREAD

**Issue by vitex** [gravitl/netmaker#863]
The installation seems to work.

```
2022/03/06 11:05:33 [netclient] joining dvm63ZwV at grpc.nm.AAA-BBB-CCC-DDD.nip.io:443
2022/03/06 11:05:33 [netclient] node created on remote server...updating configs
2022/03/06 11:05:34 [netclient] sent a node update to server for nodetrue-atomic, 7eaa0b54-7c93-4fdd-bd8b-60d59ac0d229
2022/03/06 11:05:34 [netclient] retrieving peers
2022/03/06 11:05:34 [netclient] starting wireguard
2022/03/06 11:05:34 [netclient] waiting for interface...
2022/03/06 11:05:34 [netclient] interface ready - netclient.. ENGAGE
2022/03/06 11:05:34 [netclient] joining default at grpc.nm.AAA-BBB-CCC-DDD.nip.io:443
2022/03/06 11:05:35 [netclient] node created on remote server...updating configs
2022/03/06 11:05:35 [netclient] retrieving peers
2022/03/06 11:05:35 [netclient] starting wireguard
2022/03/06 11:05:35 [netclient] waiting for interface...
2022/03/06 11:05:35 [netclient] interface ready - netclient.. ENGAGE
2022/03/06 11:05:39 [netclient] restarting netclient.service
2022/03/06 11:05:40 [netclient] joined default
```
But the client's WireGuard interfaces have no peer stanzas, so the client cannot communicate with the server.

```
interface: nm-dvm63ZwV
  public key: RHXqw8fkwM99HtsrmXVygkrlaZYyUNlovnjUhs+V7wE=
  private key: (hidden)
  listening port: 36162

interface: nm-default
  public key: 7u9hglBz/U3B3qezwhFkeuYmIceL+1Pl+Yts7aDawUo=
  private key: (hidden)
  listening port: 61821
```

**Comment by vitex** [comment]
Version v0.11.1 of `netclient join` installs peer stanzas for the mesh network interface but does not install a peer stanza for the comms network interface. Thus `netclient daemon` cannot use the comms network to receive updated parameters for peers on the mesh network.

**Comment by vitex** [comment]
Version v0.12.0 does not resolve the problem.

**Comment by afeiszli** [comment]
@vitex can you try an uninstall of the netclient first? (netclient uninstall). Do not bring down the server. Then, make sure wg interfaces are cleared locally (should be no output with "wg show") also make sure the /etc/netclient directory is empty. Then, retry the join.

**Comment by vitex** [comment]
The v0.12.1 test sequence above gave misleading results since it followed advice on Discord to use the `-p` option to specify the Listen Port; the option should have been `--port`.

The revised test sequence below seems to show that the v0.11.0 problem (the client's WireGuard interfaces have no peer stanzas, so the client cannot communicate with the server) is still present in v0.12.1.

1. Optionally install a fresh copy of Netmaker on the server node.

   ```
   docker-compose down --volumes
   wget -qO - https://raw.githubusercontent.com/gravitl/netmaker/master/scripts/nm-quick.sh | sudo bash
   ```

2. Edit the configuration of the `default` network to ensure that UDP Hole Punching is OFF and that the Default Listen Port is 54321.

3. Ensure netclient is uninstalled.

   ```
   # netclient uninstall
   [netclient] 2022-03-25 09:52:10 uninstalling netclient... 
   [netclient] 2022-03-25 09:53:44 removed systemd remnants if any existed 
   [netclient] 2022-03-25 09:53:44 uninstalled netclient 
   # wg show
   interface: <non Netmaker configuration>
   ...
   interface: <non Netmaker configuration>
   ...
   # ls /etc/netclient
   ls: cannot access '/etc/netclient': No such file or directory
   ```

4. Install a `default` network client on the server node.

   ```
   curl -sfL https://raw.githubusercontent.com/gravitl/netmaker/master/scripts/netclient-install.sh | VERSION=v0.12.1 KEY=eyJncnBj...ZSI6IiJ9 sh -
   ```

5. Display the generated WireGuard configurations for the `comms` and `default` networks.
 
   ```
   # wg show
   interface: <non Netmaker configuration>
   ...
   interface: <non Netmaker configuration>
   ...
   interface: nm-PRaNQtzW
     public key: 3zTTRgnDuB67c1EOeM9KeM23Fz9syYq6X9Lq3KsSrzg=
     private key: (hidden)
     listening port: 43830

   interface: nm-default
     public key: c6icANVs1V3SqzPhC3ObiLpGY+l2xQA1P26kU+Xyilw=
     private key: (hidden)
     listening port: 54321
   ```

    No client updates are possible since the configuration for the `comms` network does not have a peer stanza for the Netmaker server.

**Comment by vitex** [comment]
This issue was fixed in v0.13.0.
